File: java/v2.0/src/main/java/com/google/api/services/samples/dfareporting/placements/CreatePlacementGroup.java
// Copyright 2014 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.api.services.samples.dfareporting.placements;

import com.google.api.services.dfareporting.Dfareporting;
import com.google.api.services.dfareporting.model.Campaign;
import com.google.api.services.dfareporting.model.PlacementGroup;
import com.google.api.services.dfareporting.model.PricingSchedule;
import com.google.api.services.samples.dfareporting.DfaReportingFactory;

/**
 * This example creates a placement group in a given campaign. Requires the DFA site ID and campaign
 * ID in which the placement group will be created into. To create a campaign, run
 * CreateCampaign.java. To get DFA site ID, run GetSite.java.
 */
public class CreatePlacementGroup {
  private static final String USER_PROFILE_ID = "INSERT_USER_PROFILE_ID_HERE";

  private static final String CAMPAIGN_ID = "INSERT_CAMPAIGN_ID_HERE";
  private static final String DFA_SITE_ID = "INSERT_DFA_SITE_ID_HERE";
  private static final String PLACEMENT_GROUP_NAME = "INSERT_PLACEMENT_GROUP_NAME_HERE";

  public static void runExample(Dfareporting reporting, long profileId, long dfaSiteId,
      long campaignId, String placementGroupName) throws Exception {
    // Retrieve the campaign.
    Campaign campaign = reporting.campaigns().get(profileId, campaignId).execute();

    // Create a pricing schedule.
    PricingSchedule pricingSchedule = new PricingSchedule();
    pricingSchedule.setEndDate(campaign.getEndDate());
    pricingSchedule.setPricingType("PRICING_TYPE_CPM");
    pricingSchedule.setStartDate(campaign.getStartDate());

    // Create the placement group.
    PlacementGroup placementGroup = new PlacementGroup();
    placementGroup.setCampaignId(campaignId);
    placementGroup.setName(placementGroupName);
    placementGroup.setPlacementGroupType("PLACEMENT_PACKAGE");
    placementGroup.setPricingSchedule(pricingSchedule);
    placementGroup.setSiteId(dfaSiteId);

    // Insert the placement.
    PlacementGroup result = reporting.placementGroups().insert(profileId, placementGroup).execute();

    // Display the new placement ID.
    System.out.printf("Placement group with ID %d was created.%n", result.getId());
  }

  public static void main(String[] args) throws Exception {
    Dfareporting reporting = DfaReportingFactory.getInstance();

    long campaignId = Long.parseLong(CAMPAIGN_ID);
    long dfaSiteId = Long.parseLong(DFA_SITE_ID);
    long profileId = Long.parseLong(USER_PROFILE_ID);

    runExample(reporting, profileId, dfaSiteId, campaignId, PLACEMENT_GROUP_NAME);
  }
}

File: php/v2.4/examples/Placements/CreatePlacementGroup.php
<?php
/*
 * Copyright 2015 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Require the base class.
require_once dirname(__DIR__) . "/BaseExample.php";

/**
 * This example creates a placement group in a given campaign. Requires the
 * site ID and campaign ID in which the placement group will be created into.
 * To create a campaign, run CreateCampaign. To get a site ID, run GetSite.
 */
class CreatePlacementGroup extends BaseExample {
  /**
   * (non-PHPdoc)
   * @see BaseExample::getInputParameters()
   * @return array
   */
  protected function getInputParameters() {
    return array(
        array('name' => 'user_profile_id',
              'display' => 'User Profile ID',
              'required' => true),
        array('name' => 'campaign_id',
              'display' => 'Campaign ID',
              'required' => true),
        array('name' => 'site_id',
              'display' => 'Site ID',
              'required' => true),
        array('name' => 'group_name',
              'display' => 'Placement Group Name',
              'required' => true)
    );
  }

  /**
   * (non-PHPdoc)
   * @see BaseExample::run()
   */
  public function run() {
    $values = $this->formValues;

    printf(
        '<h2>Creating placement group with name "%s" under campaign ID %s</h2>',
        $values['group_name'], $values['campaign_id']
    );

    // Retrieve the campaign.
    $campaign = $this->service->campaigns->get(
        $values['user_profile_id'], $values['campaign_id']);

    $group = new Google_Service_Dfareporting_PlacementGroup();
    $group->setCampaignId($values['campaign_id']);
    $group->setName($values['group_name']);
    $group->setPlacementGroupType('PLACEMENT_PACKAGE');
    $group->setSiteId($values['site_id']);

    // Set the pricing schedule for the placement group.
    $pricing_schedule = new Google_Service_Dfareporting_PricingSchedule();
    $pricing_schedule->setEndDate($campaign->getEndDate());
    $pricing_schedule->setPricingType('PRICING_TYPE_CPM');
    $pricing_schedule->setStartDate($campaign->getStartDate());
    $group->setPricingSchedule($pricing_schedule);

    // Insert the placement group.
    $result = $this->service->placementGroups->insert(
        $values['user_profile_id'], $group
    );

    $this->printResultsTable('Placement group created.', array($result));
  }

  /**
   * (non-PHPdoc)
   * @see BaseExample::getName()
   * @return string
   */
  public function getName() {
    return 'Create Placement Group';
  }

  /**
   * (non-PHPdoc)
   * @see BaseExample::getResultsTableHeaders()
   * @return array
   */
  public function getResultsTableHeaders() {
    return array(
        'id' => 'Placement Group ID',
        'name' => 'Placement Group Name'
    );
  }
}
